### PR TITLE
Simplify code block prompts & Add doc tests in some tutorials

### PR DESCRIPTION
**Pull request by jberlanga**
This comes from discussion with @eteq & @adrn about the need to be consistent with code block prompts (e.g. `>>>` at the beginning of every line) in the docs, as well as allowing doc tests to be run on those code blocks.

### CODE REVIEW

File: docs/tutorials/plots.rst [hunk 1]
```diff
@@ -51,15 +49,23 @@ These take, at minimum, `astroplan.Observer`, `astroplan.FixedTarget` and
...
 
-Airmass versus time plots are made in the following manner::
+Airmass vs time plots are made the following way:
+
+.. code-block:: python
 
-    plot_airmass(target, observer, time)
+    >>> from astroplan.plots import plot_airmass
+
+    >>> plot_airmass(target, observer, time)
```
Comment (jberlanga) [inline docs/tutorials/plots.rst]: Looks like this line is causing the Travis CI build to fail--the doctests are looking for something named "target", which has not yet been constructed.  Only way I know to take care of this is to skip doc tests for individual code blocks until no more errors.

Comment (cdeil) [inline docs/tutorials/plots.rst]: OK to skip.

Just to offer an alternative – you could also make this a full example that works, i.e. set up target, observer and time.

When I learn a new package I usually do it by reading through the docs, executing and playing with the examples given in an IPython notebook on the side. So I prefer to have as many working examples as possible, but I also realise that it's too much setup code in some cases which can be distracting for more narrative docs.

**Comment by jberlanga** [comment]
One of the reasons that Travis CI builds are failing right now is given by [this error](https://travis-ci.org/astropy/astroplan/jobs/78016193#L673).  I'm not entirely sure where it's coming from, as I've updated my IERS table and redone the calculation a couple of times, still getting the expected output.  Thoughts?

**Comment by jberlanga** [comment]
@eteq & @adrn -- I've skipped doc tests for all the example code blocks in plots.rst (the plotting tutorial).  The reason for this is because without constructing the observers/targets/etc (outside of the chunks of code that produce the plots), doc tests are always going to fail since the code samples assume you will look elsewhere to see how observers, etc. are constructed.  

Perhaps another way to test the code for this page would be to do so for the code that actually produces the plots, which does construct observers/targets/etc.?

**Comment by bmorris3** [comment]
@jberlanga / @eteq / @adrn : Do you think this can be done by tomorrow afternoon Eastern? It'd be great to push through the release and get to work on the projects I should be working on now.

**Comment by jberlanga** [comment]
@bmorris3 -- I'm waiting on @adrn & @eteq's input on why the Travis CI builds are failing, as well as general comments.   However, anyone's input would be appreciated (@cdeil -- any comments?) 

I don't think this PR is super critical (for the 0.1 release), but I would like to see it resolved sooner rather than later, as it addresses some silly typos/inconsistencies in the tutorials as well as restoring doc-tests to some of the code examples.

**Comment by cdeil** [comment]
@jberlanga – I don't know what the cause of this small difference is between your machine and travis-ci that makes this doctest fail: https://travis-ci.org/astropy/astroplan/jobs/78017167#L673
```
     >>> sunset_tonight.iso
Expected:
    '2015-06-16 04:59:11.267'
Got:
    '2015-06-16 04:59:10.593'
```
I don't have time to look, but maybe @bmorris3 or @eteq will know ... IERS A vs B? numerical precision in set time calculation on different machines / scipy versions?

My suggestion here would be that you skip that doctest for now and rebase this, so that this can go in today for 0.1. If you think it's important, open a new issue that this should be investigated post-0.1.

Alternatively, let's just defer this PR to 0.2 and get 0.1 out the door.

@jberlanga – As you like.

**Comment by adrn** [comment]
Sorry, I didn't know you were waiting for me to comment...

I'm not thrilled with the idea of skipping this doctest -- I think this is a little bit disturbing and I don't think we want to be in the mode where people with different machines get different values with the same installation of astroplan. Maybe it's a difference in the way the interpolation works? @bmorris3 

I guess if we really want to push this out, you can skip for now but @jberlanga please try to track down where this is coming from (e.g., interpolation or whatever) and make a "known issues" page in the docs like http://docs.astropy.org/en/latest/known_issues.html with this bug on there.

**Comment by eteq** [comment]
I'm now getting a third answer for this failing test... And that makes me suspect it's IERS changes, as @cdeil suggested.  After all, that's non-deterministic so it could lead to small changes that are *real*.  Easy way to check this, though: @jberlanga, can you change all the times used here to times that are longer ago in the past?  I'd suggest going back at least a few years.  Right now most of the trouble spot is using the date ``'2015-06-15'``, which is close enough to now that the recent IERS tables are necessary and might still be changing by small amounts.  

(Note that they wouldn't be changing by ~1 sec like what we're seeing here, *but* they might be changing results enough that the interpolation scheme used to compute rise/set might be popping between different solutions given that the rise/set scheme uses a much coarser grid.)

**Comment by bmorris3** [comment]
@jberlanga: I think it sounds believable that the problem could be in varying outputs from IERS – and I agree with @eteq that maybe the best way to avoid this would be to pick a time in the more distant past.

**Comment by eteq** [comment]
@bmorris3 - it's possible, but in my experience they usually won't if you're far enough back (IERS A differs from B in the "recent" past mostly because updated data allows better measurements of the Earth's wobbles, and date from now is much more likely to impact a few months ago then decades ago).

That said, the interpolation algorithm you're using for rise/set *might* be numerically unstable to tiny changes like that at the level of seconds.  That's ok because we've already set "a few seconds" as our limit, but it means this might be an inherent problem.  In that case probably the best road is either to show something that just shows the rise/set time to minute precision, or just skip the doctest.

**Comment by jberlanga** [comment]
@astropy/astroplan-maintainers 

Alright, so here is the game plan:

1) Edit the scope of this PR to leave out doc-tests of the Summer Triangle/Plotting tutorials
2) Rebase & merge ASAP (as soon as Travis CI tests pass).
3) Create "Known Issues" page for docs in separate PR, which includes this likely IERS table/interpolation precision issue.

**Comment by bmorris3** [comment]
:+1:

**Comment by cdeil** [comment]
:+1: 

Although I think the "known issues" in the docs is overkill. There's #91, so it's a known issue.
Anyone running on a 32-bit computer will get differences of similar magnitude all over the place ... it's bad  to have this as a doctest in the first place.
(E.g. when we make a Debian package for this, it'll get tested on all kinds of machines, and we'll get fails and issue reports from 32-bit machines and then have to skip those doctests anyways.)

**Comment by eteq** [comment]
I'm ok with compromising for 0.1, but I think we absolutely need to figure out some way to get the doctests generally working.  In my experience having them off is a recipe for trouble in terms of getting the docs out-of-sync with the code.  Clearly not a problem for the *first* release, as the docs were written for that version, but we do want to figure out some way to do this.

**Comment by eteq** [comment]
Also, I don't see why you want to remove *all* the doctests for summer triangle, @jberlanga - isn't it just the one that is failing?

**Comment by cdeil** [comment]
@eteq – But as far as I see there is no good way to have robust doctests if your docs contain output like `'2015-06-16 04:59:11.267'` (see my comment on 32-bit systems above). Or course, if someone has a good idea, let's do it. But if not, this is just a fact of life: not all docs examples can be fully auto-tested.

**Comment by eteq** [comment]
@cdeil - I would say instead you skip *just* that line.  E.g.
```
>>> sunset_tonight.iso  # doctest: +SKIP
'2015-06-16 04:59:11.267'
```
That way the doctests still run to make sure the actual code up to then is valid, it just doesn't check that the final output is correct.  In my experience that gets you 99% of the way there in terms of making sure the docs don't get stale.

**Comment by cdeil** [comment]
:+1: to what @eteq said --- only skip the problematic / non-robust doctests.

**Comment by eteq** [comment]
Oh, it might also be useful to add a ``.. note::`` in the summer triangle tutorial saying that timing variations of ~seconds might show up for users who try the tutorials, but that it's nothing to worry about.  Otherwise users trying to understand these docs might get confused/frustrated that they aren't getting the same answer.

**Comment by adrn** [comment]
I'm happy with what @eteq suggests (skipping the problematic tests and adding a note)

**Comment by bsipocz** [comment]
If I can have a say I'm also with @eteq on this. Recently I've helped the sunpy guys to set up doctest on their now massive codebase, and it was amazing to see how many places the docs was out of sync with the code. Skipping a few problematic lines is way much better than getting a whole module off sync. 
And also, sometimes these doctest that picks up real regressions and corner cases that are rarely covered by unit tests.

**Comment by jberlanga** [comment]
@astropy/astroplan-maintainers -- I'll keep tests in the Summer Triangle tutorial where possible.

But I take it no "Known Issues" page for the docs anymore?

**Comment by bmorris3** [comment]
I'd agree with @cdeil that known issues should appear on the Issue tracker on GitHub. If we decided against known issues, let's remember to close #89.

**Comment by adrn** [comment]
I totally disagree about putting this in the issue tracker, but maybe for what I'm imagining "known issues" is the wrong name. A better name would be something a bit less colloquial than "gotchas." A page with things that the user may not expect but that they should be aware of, e.g., sunrise values can change unexpectedly by ~1 second because an IERS table is updated in the background.

**Comment by bmorris3** [comment]
@adrn I think @eteq suggested something like "caveats" once.

**Comment by cdeil** [comment]
I don't remember exactly what I said, but just to be clear - I'm completely OK with a "gotchas" or "caveats" or "FAQ" section in the docs that mentions things like accuracy. I even think [the whole IERS "tutoria" page](http://astroplan.readthedocs.org/en/latest/tutorials/iers.html) is more of a gotcha or caveat or background info ... it's not a tutorial that users should read when they come to astroplan and want to get some observations done.

**Comment by adrn** [comment]
:+1:

**Comment by jberlanga** [comment]
@astropy/astroplan-maintainers - Tests are passing now (with the exception of some "Accepted failures"...I'm assuming we're good with these for now?).

I did have to skip doc tests on one line to get a passing build -- See error [here](https://travis-ci.org/astropy/astroplan/jobs/78501408#L422)

    199     >>> plot_airmass(altair, subaru, time)

    UNEXPECTED EXCEPTION: ValueError('ordinal must be >= 1',)

From what I gather, this has to do with plotting dates.  I don't know why skipping just this one plotting call seems to have gotten rid of this error in the current build.  

Are we good to merge or do I need to look into this further?

**Comment by eteq** [comment]
Everything is now passing, so I'm going to go ahead and merge this.  We can always make more changes in a separate PR as time allows for making the doctests more useful
